### PR TITLE
CPDP-872 prototype alignment

### PR DESCRIPTION
**Pull request by nubz**

### CODE REVIEW

File: app/uk/gov/hmrc/cgtpropertydisposalsfrontend/models/Email.scala [hunk 1]
```diff
@@ -34,17 +33,12 @@ object Email {
 
   implicit val eq: Eq[Email] = Eq.instance(_.value === _.value)
 
-  val form: Form[Email] = {
+  val mapping: Mapping[Email] = {
     val emailRegex = "^(?=.{1,132}.$).+@.+".r.pattern.asPredicate()
```
Comment (AndyHWChung) [inline app/uk/gov/hmrc/cgtpropertydisposalsfrontend/models/Email.scala]: the regex here should be `^(?=.{3,132}$)[^@]+@[^@]+$` to be more precise

File: app/uk/gov/hmrc/cgtpropertydisposalsfrontend/models/address/Postcode.scala [hunk 2]
```diff
@@ -26,4 +29,21 @@ object Postcode {
   implicit val format: Format[Postcode] =
     implicitly[Format[String]].inmap(Postcode(_), _.value)
 
+  val mapping: Mapping[Postcode] = {
+    val postcodeRegexPredicate =
+      "^[A-Z]{1,2}[0-9][0-9A-Z]?\\s?[0-9][A-Z]{2}$|BFPO\\s?[0-9]{1,3}$".r.pattern
+        .asPredicate()
+
+    def validatePostcode(p: Postcode): ValidationResult = {
+      val postcodeWithoutSpaces = p.value.toUpperCase.replaceAllLiterally(" ", "")
```
Comment (AndyHWChung) [inline app/uk/gov/hmrc/cgtpropertydisposalsfrontend/models/address/Postcode.scala]: we need to get the rules with stripping/no stripping out spaces defined at some point

File: app/uk/gov/hmrc/cgtpropertydisposalsfrontend/models/Email.scala (hunk 1 above)
Comment (nubz) [reply app/uk/gov/hmrc/cgtpropertydisposalsfrontend/models/Email.scala]: done

**Comment by AndyHWChung** [comment]
LGTM

**Pull request merged**